CONTRIBUTING.md

```diff
@@ -31,8 +31,8 @@ maximize the chances of your PR being merged.
   breaking changes, but we make no guarantees about the length of time.
 * The breaking change policy also applies to source level extensions (e.g., filters). Code that
   conforms to the public interface documentation should continue to compile and work within the
-  deprecation window. We make no guarantees about code or deployments that rely on undocumented
-  behavior.
+  deprecation window. Within this window, a warning of depreciation should be logged. We make no
```

Comment:
I think s/depreciation/deprecation/. Also, we should probably rate limit this for some features; I wonder if we can build on the work in https://github.com/envoyproxy/envoy/pull/2783 by @gsagula to do token rate limiting of logging so we can create a LOG_EVERY_N_SECONDS or something similar.